### PR TITLE
UPBGE: Add the option to disable component execution on game start

### PR DESCRIPTION
**Pull request by lordloki**
It uses the same DEFINE as the python script auto-exec disable. Therefore, scripts and components are enable/disable using the same check at user preferences.

As blend component execution on loading as component registration or component reloading are disable by default.

Warning notifications were added to top-bar and console.

### CODE REVIEW

**Comment by BluePrintRandom** [comment]
thank you :D

**Comment by panzergame** [comment]
Nice patch, I accept it. Is there an advantage of using a separate flag for scripts and components ?

**Comment by lordloki** [comment]
Simply it is for adding more flexibility.

File: source/blender/makesrna/intern/rna_userdef.c [hunk 1]
```diff
@@ -4041,6 +4048,13 @@ static void rna_def_userdef_system(BlenderRNA *brna)
 	                         "(unsafe with blend files from an untrusted source)");
 	RNA_def_property_update(prop, 0, "rna_userdef_script_autoexec_update");
 
+	prop = RNA_def_property(srna, "use_components_auto_execute", PROP_BOOLEAN, PROP_NONE);
+	RNA_def_property_boolean_negative_sdna(prop, NULL, "flag", USER_COMPONENT_AUTOEXEC_DISABLE);
+	RNA_def_property_ui_text(prop, "Auto Run UPBGE Components",
```
Comment (panzergame) [inline source/blender/makesrna/intern/rna_userdef.c]: I think you can replace "UPBGE" by "game engine" or just "python" for "python components".

Comment (panzergame) [inline source/blender/makesrna/intern/rna_userdef.c]: You could rename "UPBGE Components" to "python components" as in the description.

**Comment by lordloki** [comment]
I reworked component auto-exec disable. Now it uses the same DEFINE as the python script auto-exec disable.

As blend component execution on loading as component registration or component reloading are disable by default.

Warning notifications were added to top-bar and console.

**Pull request merged**